weldx/schemas/weldx.bam.de/weldx/core/time_series-0.1.1.yaml

```diff
@@ -0,0 +1,124 @@
+        units: !<asdf://weldx.bam.de/weldx/tags/units/units-0.1.0> meter
+        shape: [4]
+        interpolation: step
+  -
+    - A time_series describing a sine oscillation in 3d space along the z-axis
+    - |
+      !<asdf://weldx.bam.de/weldx/tags/core/time_series-0.1.1>
+        expression: !<asdf://weldx.bam.de/weldx/tags/core/mathematical_expression-0.1.0>
+          expression: a*sin(o*t + p) + b
+          parameters:
+            a: !<asdf://weldx.bam.de/weldx/tags/units/quantity-0.1.0>
+              units: !<asdf://weldx.bam.de/weldx/tags/units/units-0.1.0> millimeter
+              value: !core/ndarray-1.0.0
+                data:
+                - [0, 0, 1]
+                datatype: int32
+                shape: [1, 3]
+            b: !<asdf://weldx.bam.de/weldx/tags/units/quantity-0.1.0> {value: 0.0, units: !<asdf://weldx.bam.de/weldx/tags/units/units-0.1.0> millimeter}
+            o: !<asdf://weldx.bam.de/weldx/tags/units/quantity-0.1.0> {value: 4.934802200544679, units: !<asdf://weldx.bam.de/weldx/tags/units/units-0.1.0> hertz * radian}
+            p: !<asdf://weldx.bam.de/weldx/tags/units/quantity-0.1.0> {value: 0, units: !<asdf://weldx.bam.de/weldx/tags/units/units-0.1.0> radian}
+        units: !<asdf://weldx.bam.de/weldx/tags/units/units-0.1.0> millimeter
+        shape: [1, 3]
+
+oneOf:
+  - type: object
+    description: |
+      Implementation for constant values.
+    properties:
+      value:
+        description: |
+          Number or n-dimensional array that is constant in time.
+        anyOf:
+          - type: number
+          - type: integer
+          - tag: "tag:stsci.edu:asdf/core/ndarray-1.*"
+      units:
+        description: |
+          Unit of the data.
+        tag: "asdf://weldx.bam.de/weldx/tags/units/units-0.1.*"
+    required: [value, units]
+
+  - type: object
+    description: |
+      Implementation for expressions.
+    properties:
+      expression:
+        description: |
+          A mathematical expression that describes the time dependent behaviour.
+        tag: "asdf://weldx.bam.de/weldx/tags/core/mathematical_expression-0.1.*"
+      units:
+        description: |
+          Resulting unit of the data when the expression is evaluated.
+        tag: "asdf://weldx.bam.de/weldx/tags/units/units-0.1.*"
+      shape:
+        description: |
+          (optional) Resulting shape of the data when the expression is evaluated.
+        type: array
+    required: [expression, units]
+
+  - type: object
+    description: |
+      Implementation for discrete data.
+    properties:
+      time:
+        description: |
+          The time axis associated with the data.
+        tag: "asdf://weldx.bam.de/weldx/tags/time/time-0.1.*"
+      units:
+        description: |
+          Units of the data.
+        tag: "asdf://weldx.bam.de/weldx/tags/units/units-0.1.*"
+      shape:
+        description: |
+          Shape of the data.
+        type: array
```

Comment:
Maybe it is a good idea to define a shape subscheme somewhere. Shapes usually have only integers as items and for our purposes sometimes strings. But floats/numbers or other types shouldn't be allowed.

Reply:
hmm yes that sounds like a good idea, should be easy to add